parser_library/src/checking/instr_operand.cpp

```diff
@@ -297,8 +304,8 @@ bool one_operand::check(
             case machine_operand_type::IMM:
                 diag = diagnostic_op::error_M122(instr_name, -boundary, boundary - 1, operand_range);
                 break;
-            case machine_operand_type::REG_IMM:
-                diag = diagnostic_op::error_M123(instr_name, -boundary, boundary - 1, operand_range);
+            case machine_operand_type::RELOC_IMM:
```

Comment:
Coupled with your other change in `operand_impls.cpp`, this code checks, whether the actual offset of relocatable symbol from the beginning of current section is within boundaries (function `is_size_corresponding_signed`). However that is wrong, the instructions we are fixing here take the difference between the relocatable symbol and the instruction adress and that should be checked for the right size.
Example: 
```
SECT CSECT
...   CODE1, which is very long
RELOC ...
... CODE2, shorter
  EXRL 1,RELOC
```
Currently, you are checking the length of CODE1 - the absolute offset from the beginning of section. However, the length of CODE2 should be checked - relative length between address of EXRL and RELOC.

To do this, we could probably introduce a special operand type whose value would be `*-<value>`. We would have to identify that we are generating specific operand for specific machine instructions...

Or extend checker API with the address of current checked instruction (however here I am not sure whether we know addresses of all instructions that are checked).

Or just remove the check and leave it for the future.

Reply:
Yes, agreed. Even I was not sure if it would give the correct difference. I think (*- <value>) for specific machine instructions would be okay for now.